lib/src/component_declaration/component_base.dart

```diff
@@ -245,7 +245,7 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component imple
     var unwrappedProps = this.unwrappedProps;
     var typedProps = _typedPropsCache[unwrappedProps];
     if (typedProps == null) {
-      typedProps = typedPropsFactory(unwrappedProps);
+      typedProps = typedPropsFactory(WarnOnModify(unwrappedProps));
```

Comment:
Since we don't want to emit this warning in prod, we can wrap these in checks so that this map is used only in dev mode:
```
typedProps = typedPropsFactory(inReactDevMode ? WarnOnModify(unwrappedProps) : unwrappedProps);
``` 
Doing it here as opposed to in WarnOnModify has the advantage of not allocating extra objects unnecessarily in prod.